fs-storage/src/file_storage.rs

```diff
@@ -62,75 +72,28 @@ where
         + Monoid<V>,
 {
     /// Create a new file storage with a diagnostic label and file path
-    pub fn new(label: String, path: &Path) -> Self {
-        Self {
+    pub fn new(label: String, path: &Path) -> Result<Self> {
```

Comment:
Also worth to mention in the comment that it's initialized with the data stored by the path, if the path exists.